File: pairwise_swap.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node* next;
} Node;

Node* push(Node* head, int data) {
    Node* newn = (Node*)malloc(sizeof(Node));
    newn->data = data;
    newn->next = head;
    return newn;
}

void pairwiseSwap(Node* head) {
    Node* cur = head;
    while (cur && cur->next) {
        int t = cur->data;
        cur->data = cur->next->data;
        cur->next->data = t;
        cur = cur->next->next;
    }
}

void printList(Node* head) {
    while (head) {
        printf("%d ", head->data);
        head = head->next;
    }
    printf("\n");
}

int main() {
    Node* head = NULL;
    for (int i = 5; i >= 1; --i) head = push(head, i); // 1..5
    printf("Original: ");
    printList(head);
    pairwiseSwap(head);
    printf("After pairwise swap: ");
    printList(head);
    return 0;
}

File: pairwise_swap.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def push(head, data):
    newn = Node(data)
    newn.next = head
    return newn

def pairwise_swap(head):
    cur = head
    while cur and cur.next:
        cur.data, cur.next.data = cur.next.data, cur.data
        cur = cur.next.next
    return head

def print_list(head):
    while head:
        print(head.data, end=" ")
        head = head.next
    print()

if __name__ == "__main__":
    head = None
    for i in range(5,0,-1):
        head = push(head, i)
    print("Original:", end=" ")
    print_list(head)
    head = pairwise_swap(head)
    print("After pairwise swap:", end=" ")
    print_list(head)
